client/src/details/SFSClientImpl.cpp

```diff
@@ -401,18 +397,18 @@ void SFSClientImpl<ConnectionManagerT>::SetCustomBaseUrl(std::string customBaseU
 }
 
 template <typename ConnectionManagerT>
-std::string SFSClientImpl<ConnectionManagerT>::GetBaseUrl() const
+std::unique_ptr<SFSUrlBuilder> SFSClientImpl<ConnectionManagerT>::MakeUrlBuilder() const
 {
     if (auto envVar = test::GetTestOverride(test::TestOverride::BaseUrl))
     {
-        return *envVar;
+        return SFSUrlBuilder::CreateFromCustomUrl(*envVar, m_instanceId, m_nameSpace, m_reportingHandler);
     }
-
-    if (m_customBaseUrl)
+    else if (m_customBaseUrl)

```

Comment:
Why change to `else if (m_customBaseUrl)` when the if block above is still doing an unconditional return from within?